pkg/build/api/types.go

```diff
@@ -411,6 +411,8 @@ const (
 	// BuildConfigLabelDeprecated was used as BuildConfigLabel before adding namespaces.
 	// We keep it for backward compatibility.
 	BuildConfigLabelDeprecated = "buildconfig"
+	// BuildConfigDeletionAnnotation is an annotation that marks a BuildConfig that is being deleted
+	BuildConfigDeletionAnnotation = "openshift.io/build-config.deletion-pending"
```

Comment:
The usage on this field looks more like "inert" than deleting, right?  This stops you from creating builds based on this bc, but no controller actually does deletion or reaping server-side based on this annotation.

We've talked about making deployments inert to keep them from doing work, this looks parallel to that.

Also, I'd like to set it to the value "true" (detection being case insensitive).  Existence is definitive, but not as explicit for users.